[Bug]: Storybook 10 broken on reload or direct access to pages with Path parameter

### Describe the bug

Since upgrading from Storybook 9.x to the latest 10.1.18, when I attempt to go directly to a story or reload a page with the path parameter, it fails to load with the following console errors:

2 globals-runtime.js:3326 Error: Minified React error #300; visit https://reactjs.org/docs/error-decoder.html?invariant=300 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at Nh (globals-runtime.js:2938:110)
    at cj (globals-runtime.js:3423:48)
    at Vk (globals-runtime.js:5237:97)
    at Uk (globals-runtime.js:4974:16)
    at Tk (globals-runtime.js:4968:28)
    at Ik (globals-runtime.js:4958:11)
    at Nk (globals-runtime.js:4748:87)
    at Gk (globals-runtime.js:4701:65)
    at J (globals-runtime.js:344:22)
    at MessagePort.R2 (globals-runtime.js:369:16)

Li @ globals-runtime.js:3326Understand this error
globals-runtime.js:5035 Uncaught Error: Minified React error #300; visit https://reactjs.org/docs/error-decoder.html?invariant=300 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at Nh (globals-runtime.js:2938:110)
    at cj (globals-runtime.js:3423:48)
    at Vk (globals-runtime.js:5237:97)
    at Uk (globals-runtime.js:4974:16)
    at Tk (globals-runtime.js:4968:28)
    at Ik (globals-runtime.js:4958:11)
    at Nk (globals-runtime.js:4748:87)
    at Gk (globals-runtime.js:4701:65)
    at J (globals-runtime.js:344:22)
    at MessagePort.R2 (globals-runtime.js:369:16)Understand this error

globals-runtime.js:5814  manager  received storybook/html/codeUpdate but was unable to determine the source of the event
error @ globals-runtime.js:5814Understand this error

globals-runtime.js:5814  manager  received storybook/instrumenter/sync but was unable to determine the source of the event


You can see this in action on this public Storybook instance: https://us-cbp.github.io/design-system/

It happens on any of the main stories (tsx files), but doesn't happen on my .mdx files like the component API Docs or Specifications or a number at the top of the left nav.

This is the web components/vite framework.

Code can be found here: https://github.com/US-CBP/design-system/tree/develop/packages/web-components

### Reproduction link

https://us-cbp.github.io/design-system/?path=/story/introduction--introduction

### Reproduction steps

Navigate to any story with a "path" URL parameter
Reload the page
It fails - check the browser console for errors.

### System

```bash
Storybook Environment Info:
│

│  System:
│  OS: Windows 11 10.0.26200
│  CPU: (24) x64 13th Gen Intel(R) Core(TM) i7-13700F
│  Binaries:
│  Node: 24.8.0 - C:\Program Files\nodejs\node.EXE
│  npm: 11.6.0 - C:\Program Files\nodejs\npm.CMD <----- active
│  Browsers:
│  Chrome: 143.0.7499.41
│  Edge: Chromium (140.0.3485.54)
│  Firefox: 145.0 - C:\Program Files\Mozilla Firefox\firefox.exe
```

### Additional context

_No response_

--- a/code/core/src/manager/components/panel/Panel.tsx
+++ b/code/core/src/manager/components/panel/Panel.tsx
@@ -67,11 +67,17 @@ const Section = styled.section({
   flexDirection: 'column',
 });
 
+// Ensures we don't cause difficult-to-debug hooks violations
+// whether addon authors pass a React component ctor or anonymous function.
+function renderChild(RenderProp: Addon_BaseType['render']) {
+  return <RenderProp active={true} />;
+}
+
 // Avoids crashes due to rules of hooks.
 const PreRenderAddons = ({ panels }: { panels: Record<string, Addon_BaseType> }) => {
   return Object.entries(panels).map(([k, v]) => (
     <StatelessTabPanel key={k} name={k} hasScrollbar={false}>
-      <TabErrorBoundary key={k}>{v.render({ active: true })}</TabErrorBoundary>
+      <TabErrorBoundary key={k}>{renderChild(v.render)}</TabErrorBoundary>
     </StatelessTabPanel>
   ));
 };

